vignettes/vcr.Rmd

```diff
@@ -1,191 +1,172 @@
 
-The main use case is for unit tests, but you can use it outside of the unit test use case.
-
-`vcr` works with the `crul`, `httr` and `httr2` HTTP request packages.
-
-Check out the [HTTP testing book](https://books.ropensci.org/http-testing/) for a lot more documentation on `vcr`, `webmockr`, and `crul`, and other packages.
-
-## Basic usage
-
-### In tests
+```{r setup}
+library(vcr)
+library(testthat)
+library(httr2, warn.conflicts = FALSE)
 
-In your tests, for whichever tests you want to use `vcr`, wrap them in a `vcr::use_cassette()` call like:
+httpbin <- webfakes::local_app_process(webfakes::httpbin_app())
+```
 
-```r
-library(testthat)
-vcr::use_cassette("rl_citation", {
-  test_that("my test", {
-    aa <- rl_citation()
-
-    expect_type(aa, "character")
-    expect_match(aa, "IUCN")
-    expect_match(aa, "www.iucnredlist.org")
-  })
-})
+```{r}
+#| include: false
+# Don't reocrd any cassette for this vignette so they're always created afresh
+vcr_configure(dir = tempdir())
 ```
 
-OR put the `vcr::use_cassette()` block on the inside, but put `testthat` expectations outside of
-the `vcr::use_cassette()` block:
+## Getting started
 
-```r
-library(testthat)
+The central metaphor of vcr is a video cassette.
+This is what records your HTTP **interactions**, the pairs of HTTP request and response.
+You **insert** a cassette to begin the replay (or start the recording) and **eject** it when you're done.
+To use vcr in a test, include a call to `vcr::local_cassette()`: it inserts the cassette and automatically ejects it when the test is done.
+
+```{r}
+#| label: get-record
 test_that("my test", {
-  vcr::use_cassette("rl_citation", {
-    aa <- rl_citation()
-  })
+  vcr::local_cassette("get")
+  
+  req <- request(httpbin$url("/get"))
+  resp <- req_perform(req)
+  json <- resp_body_json(resp)
 
-  expect_type(aa, "character")
-  expect_match(aa, "IUCN")
-  expect_match(aa, "www.iucnredlist.org")
+  expect_named(json, c("args", "headers", "origin", "path", "url"))
 })
 ```
 
-Don't wrap the `use_cassette()` block inside your  `test_that()` block with `testthat` expectations inside the `use_cassette()` block, as you'll only get the line number that the `use_cassette()` block starts on on failures.
-
-The first time you run the tests, a "cassette" i.e. a file with recorded HTTP interactions, is created at `tests/fixtures/rl_citation.yml`.
-The times after that, the cassette will be used.
-If you change your code and more HTTP interactions are needed in the code wrapped by `vcr::use_cassette("rl_citation"`, delete `tests/fixtures/rl_citation.yml` and run the tests again for re-recording the cassette.
-
-### Manual editing
+The first time you run the test it will record all request and responses in a **cassette**, a YAML file that lives in `test/testthat/_vcr`.
+In future runs, it will replay those responses, freeing your test from the vagaries of the internet, making it both faster and more reliable.
```

Comment:
Would it help or not help to compare this to snapshot tests, where the snapshot is recorded the first time you run the tests? 

Or, should there be a mention of the word "cache" somewhere? Cassettes are a bit like a cache. The former vignette text had "cached".

Reply:
Good idea. I'll make those connections explicit.